TypeScript/TS/tscript.ts

```diff
@@ -0,0 +1,382 @@
+var buttonText:string|number, tempVarTrigo:string|number;
+var result = document.querySelector("#result")! as HTMLInputElement;
+var resultToShow = document.querySelector(".result")! as HTMLInputElement;
+var resultValue:string|number = "";
+var buttons= document.querySelectorAll("button")! ;
+var buttonLength = document.querySelectorAll("button").length;
+var degButton = document.querySelector(".deg")!;
+var radButton = document.querySelector(".rad")!;
+var memorySave:number|string;
+
+//for radian to degree conversion
+function degFunction() {
+    return +resultValue * (Math.PI / 180);
+
+}
+//for parenthesis loop hole:invalid syntax
+let brackets:number;
+let openingBracket = document.querySelector(".open_parenthesis")!;
+let closingBracket = document.querySelector(".closing_parenthesis")!;
+openingBracket.addEventListener("click", () => {
+    brackets++;
+})
+closingBracket.addEventListener("click", () => {
+    brackets--;
+})
+
+//Event listeners and evaluation for all the buttons
+
+for (var i = 0; i < buttonLength; i++) {
+    buttons[i].addEventListener("click", function(event) {
+        buttonText = (event.target as HTMLElement).textContent !;
```

Comment:
Remove space between `textContent` and `!`.
Create a union type of all possible values of `buttonText` for switch case.